src/pages/index.js

```diff
@@ -73,35 +102,137 @@ function handleImageClick(title, link) {
 }
 
 function makeCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardTemplate,
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike
+  );
   return card.generateCard();
 }
 
+function checkIfLiked(ifLiked, element) {
+  if (ifLiked === true) {
+    element
+      .querySelector(".heart-button")
+      .classList.toggle("heart-button_active");
+  }
+}
+
+function handleDeleteClick(cardId) {
+  console.log(cardId);
+  deleteConfirmModal.open(cardId);
+}
+
 // VALIDATION
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const cardFormValidator = new FormValidator(validationSettings, cardAddForm);
+const avatarFormValidator = new FormValidator(
+  validationSettings,
+  avatarEditForm
+);
 
+avatarFormValidator.enableValidation();
 editFormValidator.enableValidation();
 cardFormValidator.enableValidation();
 
 /* EVENT HANDLERS */
 
 function handleProfileEditSubmit({ name, badge }) {
-  user.setUserInfo(name, badge);
-  editProfileModal.close();
+  editProfileModal.setLoading(true, "Saving...");
+  api
+    .updateProfile({ name, badge })
+    .then((message) => {
+      console.log(message);
+      user.setUserInfoOnSubmit(name, badge);
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false, "Save");
+      editProfileModal.close();
+    });
 }
 
 function handleCardAddSubmit({ name, cardUrl }) {
-  section.addItem(
-    makeCard({
-      title: name,
-      link: cardUrl,
+  newCardModal.setLoading(true, "Saving...");
+  api
+    .addCard({ name, cardUrl })
+    .then((card) => {
+      console.log(card);
+      cardSection.addItem(makeCard(card));
     })
-  );
-  newCardModal.close();
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      newCardModal.setLoading(false, "Save");
+      newCardModal.close();
+    });
+}
+
+function handleCardDeleteSubmit(card) {
+  console.log(card._id);
+  deleteConfirmModal.setLoadingConfirm(true, "Saving...");
+  api
+    .deleteCard(card._id)
+    .then((message) => {
+      console.log(message);
+      card.domDeleteCard();
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      deleteConfirmModal.setLoadingConfirm(true, "Yes");
+      deleteConfirmModal.close();
+    });
+}
+
+function handleAvatarChangeSubmit(url) {
+  console.log(url.avatarUrl);
+  avatarChangeModal.setLoading(true, "Saving...");
+  api
+    .updateAvatar(url.avatarUrl)
+    .then((res) => {
+      console.log(res);
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      avatarChangeModal.setLoading(false, "Save");
+      user.setAvatarPic(url.avatarUrl);
```

Comment:
`      user.setAvatarPic(url.avatarUrl);
`

You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error, it shouldn’t change